File: spark/learning-spark-lightning-fast-big-data-analysis/python/line_filter.py
# coding: utf-8

import findspark
findspark.init()

import pyspark

from pyspark import SparkContext
from pyspark import SparkConf

conf = SparkConf().setAppName('LineFilter').setMaster('local[*]')
sc = SparkContext.getOrCreate(conf)

lines = sc.textFile("/Users/me/Code/spark-learning/learning-spark-lightning-fast-big-data-analysis/pom.xml")
xml_lines = lines.filter(lambda line: 'xml' in line)
print xml_lines.first()

File: spark/learning-spark-lightning-fast-big-data-analysis/jvm/src/main/scala/redocpot/spark/LineFilter.scala
package scala.redocpot.spark

import org.apache.spark.SparkConf
import org.apache.spark.SparkContext
import org.apache.spark.SparkContext._

object LineFilter {
  def main(args : Array[String]) : Unit = {
    val conf = new SparkConf().setAppName("LineFilter").setMaster("local[*]")
    val sc = new SparkContext(conf)
    val textFile = sc.textFile("/Users/me/Code/spark-learning/learning-spark-lightning-fast-big-data-analysis/pom.xml")
    val xmlLines = textFile.filter(line => line.contains("xml"))
    println(xmlLines.first)
    sc.stop
  }
}
